File: app/src/main/java/fr/corenting/edcompanion/network/retrofit/InaraRetrofit.java
package fr.corenting.edcompanion.network.retrofit;

import fr.corenting.edcompanion.models.apis.Inara.InaraProfileResponse;
import fr.corenting.edcompanion.models.apis.Inara.InaraProfileRequestBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface InaraRetrofit {
    @POST("inapi/v1/")
    Call<InaraProfileResponse> getProfile(@Body InaraProfileRequestBody body);
}

File: app/src/main/java/fr/corenting/edcompanion/network/retrofit/InaraRetrofit.kt
package fr.corenting.edcompanion.network.retrofit

import fr.corenting.edcompanion.models.apis.Inara.InaraProfileRequestBody
import fr.corenting.edcompanion.models.apis.Inara.InaraProfileResponse
import retrofit2.http.Body
import retrofit2.http.POST

interface InaraRetrofit {
    @POST("inapi/v1/")
    suspend fun getProfile(@Body body: InaraProfileRequestBody): InaraProfileResponse
}